submissions/YuraZagor/tjsw/index.js

```diff
@@ -0,0 +1,154 @@
+const manNamesArr = ['Vasil', 'Yurij', 'Zahar', 'Taras', 'Olexiy', 'Oleg', 'Mykola', 'Sashko', 'Sergij', 'Bohdan']
+const womanNamesArr = ['Hanna', 'Helena', 'Vasilisa', 'Yulya', 'Oksana', 'Olena', 'Olga', 'Myroslava', 'Svitlana', 'Bohdana']
+const animalNamesArr = ['Joy', 'Happy', 'Lucky', 'Puffy']
+
+let catSay = 'meau';
+let dogSay = 'woufff';
+let humanSay = 'Glory to Ukraine!';
+
+function randomSelect(arr){
+  return arr[Math.floor(Math.random()*arr.length)]
+}
+let manName = randomSelect(manNamesArr)
+let womanName = randomSelect(womanNamesArr)
+let animalName = randomSelect(animalNamesArr)
+
+const inhabitants = [
+  { species: 'human', 
+    gender: 'male',
+    name: 'Vasil',
+    saying: 'Slava natsiyi',
+    friend: 'Hanna',
+    legs: 2,
+    hands: 2,
+  },
+  { species: 'human', 
+    gender: 'female',
+    name: 'Hanna',
+    saying: 'Smert vorogam',
+    friend: 'Vasil',
+    legs: 2,
+    hands: 2,
+  },
+];
+
+class Inhabitant {  
+  constructor(species, name, saying, friend, gender, habitat=inhabitants){
+    this.species=species;
+    this.name=name;
+    this.saying = saying
+    this.friend=friend;
+    this.gender=gender;
+    this.habitat = habitat
+  }
+  accountInhabitants(){
+    this.habitat.push(this);
+  }
+  commonProperties(){
+    const { species, name, saying, friend, gender} = this
+    const commonProperties = [species, name, saying, friend, gender]
+    return commonProperties    
+  }
+};
+
+class Animal extends Inhabitant{
+  constructor( species, name, saying, friend, gender, paws, tail ){
+    super(species, name, saying, friend, gender);
```

Comment:
What `super` is and what it does?

Reply:
it saves me from repetedly asigning arguments as this.args by calling parent-class constructor I believe. Also, there is magic about 'this' involved, - we must call 'super' before using this keyword for activation of 'derived class' property

Reply:
Sorry, didn't think I have to answer it given I can google the answer. Anyways, as I understand it:
1. super() if a callback call to parent class, which is a function object itself. As an object, it has props, one of them is prototype prop, that itself is an object.
2. when called, it creates a new object, which gets the __proto__ prop pointing to it's prototyping function object (parent) props (methods in object-model of a function are also props). 
3. Creating-a-new-obj+assigning-'this.'   model is located in first parent, and is not duplicated is children, they get a prop 'derived' which instructs the function to look deeper in the prototyping tree for this method. That is how new obj 'is aware' of all props and methods of all it's ancestors, if there's more than one.

Can do the same from non-object, but functional point-of-view, but that would be kinda strange in OOP I guess